### PR TITLE
Feature: web3-react connector

### LINKED ISSUE #121: Add web3react connector
Multiple projects are asking for a way for their apps to be used as a safe app via web3-react. It seems like in order to make it possible we need to create a connector based on https://github.com/NoahZinsmeister/web3-react/blob/v6/packages/abstract-connector/src/index.ts

### PR DESCRIPTION
**Pull request by mikheevm**
This PR:
- Adds `@gnosis.pm/safe-apps-web3-react` package that includes a web3-react connector and a helper hook
- Also release safe-apps-provider@0.2.2 package with bugfixes. For me the npm dep didn't have a dist folder, this could be a local issue, but I published the package with no major changes
- Moves LICENSE from safe-apps-sdk package to repo root
- Closes #121 
- Removes `safe-apps-ethers-provider` in favor of generic `safe-apps-provider` that is compatible with both web3.js and ethers
- Fixes type of request function as it was not compatible with ethers.js

Tested on https://github.com/sushiswap/sushiswap-interface and works fine

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,15 +9,16 @@ You can find more resources on Safe Apps in the [Gnosis Safe Developer Portal](h
...
+| [safe-apps-react-sdk](https://github.com/gnosis/safe-apps-sdk/tree/master/packages/safe-apps-react-sdk)                                                                                                                        | A wrapper of safe-apps-sdk with helpful hooks                                                               |
+| [safe-apps-sdk](https://github.com/gnosis/safe-apps-sdk/tree/master/packages/safe-apps-sdk) [![npm version](https://badge.fury.io/js/%40gnosis.pm%2Fsafe-apps-sdk.svg)](https://badge.fury.io/js/%40gnosis.pm%2Fsafe-apps-sdk) | JavaScript SDK                                                                                              |
+| [safe-apps-ethers-provider](https://github.com/gnosis/safe-apps-sdk/tree/master/packages/safe-apps-ethers-provider)                                                                                                            | An ethers.js provider and signer to be used with ethers.js contract instances                               |
```
Comment (rmeissner) [inline README.md:17]: Should we remove that one?

File: packages/safe-apps-web3-react/src/useSafeAppConnection.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { useWeb3React } from '@web3-react/core';
+import { SafeAppConnector } from './connector';
+
+function useSafeAppConnection(connector: SafeAppConnector): boolean {
+  const { activate, active } = useWeb3React(); // specifically using useWeb3ReactCore because of what this hook does
```
Comment (rmeissner) [inline packages/safe-apps-web3-react/src/useSafeAppConnection.ts:6]: Is this comment still required? (it mentions another method)

File: packages/safe-apps-web3-react/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+### Helper hook
+
+You can use our helper hook to automatically connect to a safe, it will automatically connect to the Safe if it detects that it's loaded in Safe App context:
+
+```js
+import { useSafeAppConnection } from '@gnosis.pm/safe-apps-web3-react';
+
+const App = () => {
+  const triedToConnectToSafe = useSafeAppConnection();
```
Comment (rmeissner) [inline packages/safe-apps-web3-react/README.md:29]: Does this require a parameter (connector)?

https://github.com/gnosis/safe-apps-sdk/pull/122/files#diff-6fdbd8820b0ce753e3cd26c61685a0fd2e5324e813f1bd1e42fdee690f47dcfbR5

File: packages/safe-apps-web3-react/src/useSafeAppConnection.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  const [tried, setTried] = React.useState(false);
+
+  React.useEffect(() => {
+    connector.isSafeApp().then((loadedInSafe) => {
+      if (loadedInSafe) {
+        activate(connector, undefined, true).catch(() => {
```
Comment (rmeissner) [inline packages/safe-apps-web3-react/src/useSafeAppConnection.ts:12]: not really required, but maybe makes it easier to understand: Should we comment tat `activate` flips the `active` flag and therefore it is not required to call `setTried(true)` on success?

**Review by rmeissner — APPROVED** [review]
Looks good, minor comments nothing blocking.

File: README.md (hunk 1 above)
Comment (mikheevm) [reply README.md:17]: We should, I thought I'd do it in a separate PR so its easier to review

**Review by rmeissner — APPROVED** [review]

**Pull request merged**